pkg/ocm/ocm.go

```diff
@@ -156,3 +184,40 @@ func (c *Controller) updateSecret(s *v1.Secret, ocmData *ScaResponse) (*v1.Secre
 	}
 	return s, nil
 }
+
+// requestSCAWithExpBackoff queries OCM API with exponential backoff and returns
+// an error only in case of an HTTP error other than 404 received from the OCM API.
+// Data return value still can be an empty array in case of HTTP 404 error.
+func (c *Controller) requestSCAWithExpBackoff(endpoint string) ([]byte, error) {
+	bo := wait.Backoff{
+		Duration: 5 * time.Minute,
+		Factor:   2,
+		Jitter:   0,
+		Steps:    status.OCMAPIFailureCountThreshold,
+	}
+	var data []byte
+	err := wait.ExponentialBackoff(bo, func() (bool, error) {
+		var err error
+		data, err = c.client.RecvSCACerts(c.ctx, endpoint)
+		if err != nil {
+			// don't try again in case it's not an HTTP error - it could mean we're in disconnected env
+			if !insightsclient.IsHttpError(err) {
+				klog.Errorf("Failed to request the SCA certs: %v", err)
+				return true, nil
+			}
+			httpErr := err.(insightsclient.HttpError)
+			// don't try again in case of 404
+			if httpErr.StatusCode == http.StatusNotFound {
+				return true, nil
+			}
+			klog.Errorf("%v. Trying again in %s", httpErr, bo.Step())
+			return false, nil
+		}
+		return true, nil
+	})
+	// exp. backoff timeouted -> error
+	if err != nil {
+		return nil, fmt.Errorf("timed out waiting for the successful response from %s", endpoint)
+	}
+	return data, nil
```

Comment:
This method is little bit complicated (especially due the fact that 404 and offline env. are not considered as errors from the "degraded" perspective) compared to counting the threshold in the [controllerstatus](https://github.com/openshift/insights-operator/blob/master/pkg/controllerstatus/controllerstatus.go#L32) and the checking it in the `status.go` (similarly as [here](https://github.com/openshift/insights-operator/blob/master/pkg/controller/status/status.go#L164))

The approach with exp. backoff is more "agressive":
- if HTTP error other than 404 then try again in 5 min (if this is still true then try again in 10 min, 20min, 40min => mark as degraded)

Compared to the other approach:
- if HTTP error other than 404 then only increase the count and do nothing
-  try again in default period (which is 8h in this case) if there's still an error then increase the count again
- if count higher than some threshold (let's say 2 or 3) then mark IO as degraded

WIth the first approach we can go degraded in ~1h15min, with the second one it would be after 16h-24h. I am not sure what is better TBH. The certs are valid for a long time (1 year by default), but on the other hand, if the content changes then you may want to update the certs quickly.

Reply:
I updated the initial backoff step to `c.configurator.Config().OCMConfig.Interval / 32` (that is 15 minutes by default) so the times are bit longer than mentioned above.